File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    const answer = [];
    
    for (const target of targets) { // "ABCD"
        let cnt = 0;
        
        for (let str of target) {   // A
            let idx = 100;  // 인덱스는 99 이하이므로 100으로 초기값 설정
            let isNotKey = false;   // 목표 문자열 작성 불가
            
            for (let key of keymap) {
                if (key.includes(str)) {
                    isNotKey = true;
                    idx = Math.min(idx, (key.indexOf(str) + 1));
                }
            }
            
            if (!isNotKey) {
                cnt = -1;
                break;
            } else {
                cnt += idx;
            }
        }
        
        answer.push(cnt);
    }
    
    return answer;
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    
    for target in targets:
        # ABCD
        cnt = 0
        
        for char in target:
            # A
            idx = 100
            is_not_key = False
            
            # 각 keymap에서 char의 인덱스 확인, 최소값을 idx에 저장. 없을 경우 바로 -1 리턴
            for key in keymap:
                if char in key:
                    is_not_key = True;
                    idx = min(idx, key.index(char) + 1)
            
            if is_not_key:
                cnt += idx
            else:
                cnt = -1
                break
            
        answer.append(cnt)
        
    return answer